nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/ScanKudu.java

```diff
@@ -0,0 +1,392 @@
+            .defaultValue("500")
+            .required(true)
+            .addValidator(StandardValidators.createLongValidator(1, 100000, true))
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All FlowFiles are routed to this relationship")
+            .build();
+
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("The original input file will be routed to this destination, even if no rows are retrieved based on provided conditions.")
+            .build();
+
+    protected static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it is not able to read from Kudu")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(KUDU_MASTERS);
+        properties.add(KERBEROS_CREDENTIALS_SERVICE);
+        properties.add(KUDU_OPERATION_TIMEOUT_MS);
+        properties.add(KUDU_KEEP_ALIVE_PERIOD_TIMEOUT_MS);
+        properties.add(TABLE_NAME);
+        properties.add(PREDICATES);
+        properties.add(PROJECTED_COLUMNS);
+        properties.add(BATCH_SIZE);
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    static final Set<Relationship> relationships;
+    static {
+        Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_ORIGINAL);
+        rels.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(rels);
+    }
+
+    static final String KUDU_TABLE_ATTR = "kudu.table";
+
+    static final String KUDU_ROWS_COUNT_ATTR = "kudu.rows.count";
+
+    protected KuduTable kuduTable;
+    protected List<String> projectedColumnNames;
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) throws LoginException {
+        createKuduClient(context);
+    }
+
+    @Override
+    public void trigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile fileToProcess = null;
+        if (context.hasIncomingConnection()) {
+            fileToProcess = session.get();
+
+            // If we have no FlowFile, and all incoming connections are self-loops then we can continue on.
+            // However, if we have no FlowFile and we have connections coming from other Processors, then
+            // we know that we should run only if we have a FlowFile.
+            if (fileToProcess == null && context.hasNonLoopConnection()) {
+                return;
+            }
+        }
+
+        projectedColumnNames = Arrays.asList(context.getProperty(PROJECTED_COLUMNS).getValue().split(","));
```

Comment:
`PROJECTED_COLUMNS` is not required but this will throw a `NullPointerException` if it is not set.
I'd suggest changing this to something like this:
```java
        projectedColumnNames = Optional.ofNullable(context.getProperty(PROJECTED_COLUMNS))
                .map(PropertyValue::getValue)
                .map(value -> value.split(","))
                .map(Arrays::asList)
                .orElseGet(() -> Collections.emptyList());
```